addon/utils/store.js

```diff
@@ -21,11 +21,30 @@ export default EmberObject.extend({
   },
 
   initialsFor(properties) {
-    let key = hash(properties);
+    let hashableProp = this._ensureHashable(properties);
+    let key = md5(hashableProp);
     return this.get('cache')[key] || this._create(key, properties);
   },
 
   _create(key, properties) {
     return this.get('cache')[key] = this.get('generator').generate(properties);
+  },
+
+  _ensureHashable(properties) {
+    if (typeOf(properties) !== 'object') {
+      // this should never happen, but just in case
+      return properties.toString();
+    }
+
+    let pKeys = Object.keys(properties).sort();
+    let result = [];
+    pKeys.forEach((key) => {
+      let value = properties[key];
+      if (typeOf(value) !== 'string') {
```

Comment:
we should definitely have more tests for every of the possible case with string number object array function and custom class

Reply:
I will take care of that

Reply:
Done in #44

Reply:
sure, np